includes/class-amp-theme-support.php

```diff
@@ -829,7 +828,21 @@ protected static function ensure_required_markup( DOMDocument $dom ) {
 			) );
 			$head->insertBefore( $meta_viewport, $meta_charset->nextSibling );
 		}
-
+		// Prevent schema.org duplicates.
+		$schema_org_script = null;
+		foreach ( $head->getElementsByTagName( 'script' ) as $script ) {
+			if ( 'application/ld+json' === $script->getAttribute( 'type' ) && preg_match( '/{"@context":"https?:[\\\]\/[\\\]\/schema\.org/', $script->nodeValue ) ) {
+				$schema_org_script = $script->nodeValue;
```

Comment:
Since the value isn't being used, it may be better to just store a flag like `$has_schema_org_metadata = true`.

Reply:
Thanks, it is true, i made the corresponding correction.